lib/matplotlib/colorbar.py

```diff
@@ -592,6 +597,7 @@ def _ticker(self):
         formatter.set_data_interval(*intv)
 
         b = np.array(locator())
+        self._tick_data_values = b
```

Comment:
It looks like you are close, but shouldn't you save the b values after all potential modifications?  See the following lines.